### PR TITLE
add aggregateOnly and rawOnly config and constraint

### LINKED ISSUE #6: Constraint: AggregateOnly / RawOnly
- [x] AggregateOnly
- [x] RawOnly

### PR DESCRIPTION
**Pull request by felixcodes**
fix #6

- add aggregateOnly config
- add rawOnly config
- add aggregateOnly constraint
- add rawOnly constraint
- add tests for both constraints

### CODE REVIEW

**Comment by felixcodes** [comment]
Travis is failing due to a network error.

File: src/constraint/spec.ts [hunk 1]
```diff
@@ -102,6 +102,21 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
     }
   },
   {
+    name: 'aggregateOnly',
+    description: 'Omit raw plots.',
+    properties: [Property.AGGREGATE, Property.AUTOCOUNT],
+    allowEnumSpecForProperties: true,
+    strict: false,
+    satisfy: (specM: SpecQueryModel, schema: Schema, opt: QueryConfig) => {
+      if (opt.aggregateOnly) {
```
Comment (kanitw) [inline src/constraint/spec.ts]: You don't need to check for this.  Our generation would trigger this constraints only if it is true.

File: src/constraint/spec.ts [hunk 2]
```diff
@@ -429,6 +444,21 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
...
     }
   },
+  {
+    name: 'rawOnly',
+    description: 'Omit aggregate plots.',
+    properties: [Property.AGGREGATE, Property.AUTOCOUNT],
+    allowEnumSpecForProperties: true,
+    strict: false,
+    satisfy: (specM: SpecQueryModel, schema: Schema, opt: QueryConfig) => {
+      if (opt.rawOnly) {
```
Comment (kanitw) [inline src/constraint/spec.ts]: You don't need to check for this.  Our generation would trigger this constraints only if it is true.

File: test/constraint/spec.test.ts [hunk 3]
```diff
@@ -33,6 +33,30 @@ describe('constraints/spec', () => {
     }
   });
 
+  describe('aggregateOnly', () => {
+    const CONFIG_WITH_AGGREGATE_ONLY = extend({}, DEFAULT_QUERY_CONFIG, {aggregateOnly: true});
+
+    it('should return false if there is raw data when aggregateOnly config is turned on', () => {
```
Comment (kanitw) [inline test/constraint/spec.test.ts]: cut "when", because it doesn't matter for constraint test... same for others...

**Comment by kanitw** [comment]
I suggest that you open vega-editor and choose 1D and add omitRaw/Aggregate and see if the output makes sense.  If so, just copy the spec and assert the sensible result.  Should be easy to write. ;)

**Pull request merged**